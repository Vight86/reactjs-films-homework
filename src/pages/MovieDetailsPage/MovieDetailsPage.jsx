import React, { useState } from 'react';
import styles from './MovieDetailsPage.scss';
import Cover from '../../components/Cover/index';
import Title from '../../components/Title/index';
import GenreList from '../../components/GenreList/index';
import Runtime from '../../components/Runtime/index';
import Rating from '../../components/Rating/index';
import Button from '../../components/Button/index';
import image from '../../components/MovieCard/movieStab.jpg';
import MovieSynopsis from '../../components/MovieSynopsis/index';

const stab = {
  image,
  title: 'Independance day',
  runtime: 240,
  rating: 1.9,
  genre: ['scifi', 'comedy'],
  info: `There are growing dangers in
    the wizarding world of 1926 New York.
    Something mysterious is leaving a path of destruction in the streets,
    threatening to expose the wizarding`,
};

const MovieDetailsPage = () => {
  const [isSynopsisShown, setIsSynopsisShown] = useState(false);
  const showMovieSynopsis = () => {
    setIsSynopsisShown(!isSynopsisShown);
  };

  return (
    <section className={styles.movieDetails}>
      <Cover className="inHeroSection">
        <img src={stab.image} alt={stab.title} />
      </Cover>
      <div className={styles.movieInfo}>
        <div className={styles.movieInfo__left}>
          <Title className="primary">{stab.title}</Title>
          <div className={styles.movieSubInfo}>
            <GenreList className="primary">{stab.genre}</GenreList>
            <Runtime>{stab.runtime}</Runtime>
          </div>
          <Rating className="primary" withStars>{stab.rating}</Rating>
        </div>
        <MovieSynopsis className="primary" isShown={isSynopsisShown}>{stab.info}</MovieSynopsis>
        <div className={styles.movieInfo__right}>
          <Button className="secondary">Watch Now</Button>
          <Button className="secondary" onClick={showMovieSynopsis}>View Info</Button>
        </div>
      </div>
    </section>
  );
};

export default MovieDetailsPage;
