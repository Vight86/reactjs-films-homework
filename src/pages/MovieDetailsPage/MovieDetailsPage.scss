@import '../../utils/variables';

.movieDetails {
  position: relative;
  width: 100%;
}

.movieInfo {
  position: absolute;
  bottom: 0;
  left: 0;
  display: grid;
  width: 100%;
  box-sizing: border-box;
  padding-right: $pr;
  padding-bottom: 3%;
  padding-left: $pl;
  grid-template-columns: minmax(0, 1fr) 45em;

  &__right {
    display: flex;
    align-self: end;
    justify-content: space-between;

    button {
      margin: 0;
    }
  }
}
