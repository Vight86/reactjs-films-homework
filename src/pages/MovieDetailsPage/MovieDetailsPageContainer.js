import React, { useEffect } from 'react';
import propTypes from 'prop-types';
import {
  shallowEqual, useDispatch, useSelector,
} from 'react-redux';
import { useParams } from 'react-router-dom';

import { selectQuery } from '../../modules/query/index';
import { selectMovieDetails, loadMovieDetails } from '../../modules/movies/index';

import MovieDetailsPage from './MovieDetailsPage';
import Preloader from '../../components/Preloader/index';

const MovieDetailsPageContainer = ({ isModalOpened, handleModalToggle }) => {
  const dispatch = useDispatch();

  const { id } = useParams();
  const {
    basePath, apiKey, lang,
  } = useSelector(selectQuery);

  useEffect(() => {
    const fetchMovieDetailsUrl = `${basePath}/movie/${id}?api_key=${apiKey}&language=${lang}`;
    dispatch(loadMovieDetails(fetchMovieDetailsUrl));
    window.scrollTo(0, 0);
  }, [id]);

  const currentMovieDetails = useSelector(selectMovieDetails, shallowEqual);

  return (
    currentMovieDetails.title
      ? (
        <MovieDetailsPage
          isModalOpened={isModalOpened}
          handleModalToggle={handleModalToggle}
          id={id}
          currentMovieDetails={currentMovieDetails}
        />
      )
      : <Preloader />
  );
};

MovieDetailsPageContainer.propTypes = {
  isModalOpened: propTypes.bool.isRequired,
  handleModalToggle: propTypes.func.isRequired,
};

export default MovieDetailsPageContainer;
