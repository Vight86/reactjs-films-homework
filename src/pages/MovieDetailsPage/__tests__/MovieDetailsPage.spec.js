import React from 'react';
import Renderer, { act } from 'react-test-renderer';
import MovieDetailsPage from '../index';

const setUp = () => {
  const renderer = Renderer.create(<MovieDetailsPage />);
  return renderer;
};

describe('render MovieDetailsPage component', () => {
  it('render component correctly', () => {
    const result = setUp();
    expect(result.toJSON()).toMatchSnapshot();
  });

  it('render <View Info> button clicks correctly', () => {
    const result = setUp();
    const viewInfoButton = result.root.findByProps({ children: 'View Info' });
    act(() => viewInfoButton.props.onClick());
    expect(result.toJSON()).toMatchSnapshot();
    act(() => viewInfoButton.props.onClick());
    expect(result.toJSON()).toMatchSnapshot();
  });
});
