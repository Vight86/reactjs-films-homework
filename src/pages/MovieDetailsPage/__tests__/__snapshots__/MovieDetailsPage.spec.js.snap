// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render MovieDetailsPage component render <View Info> button clicks correctly 1`] = `
<section
  className="movieDetails"
>
  <figure
    className="inHeroSection"
  >
    <img
      alt="Independance day"
      src={
        Object {
          "canInstrument": true,
          "createTransformer": [Function],
          "getCacheKey": [Function],
          "process": [Function],
        }
      }
    />
  </figure>
  <div
    className="movieInfo"
  >
    <div
      className="movieInfo__left"
    >
      <h1
        className="primary"
      >
        Independance day
      </h1>
      <div
        className="movieSubInfo"
      >
        <ul
          className="primary"
          style={
            Object {
              "order": null,
            }
          }
        >
          <li>
            scifi
          </li>
          <li>
            comedy
          </li>
        </ul>
        <span
          className="runtime"
        >
          4h
        </span>
      </div>
      <div
        className="rating"
      >
        <span
          className="stars"
          style={
            Object {
              "width": "10.4em",
            }
          }
        />
        <span
          className="primary"
        >
          1.9
        </span>
      </div>
    </div>
    <p
      className="primary active"
    >
      <span>
        There are growing dangers in
    the wizarding world of 1926 New York.
    Something mysterious is leaving a path of destruction in the streets,
    threatening to expose the wizarding
      </span>
    </p>
    <div
      className="movieInfo__right"
    >
      <button
        className="secondary"
        onClick={[Function]}
        type="button"
      >
        Watch Now
      </button>
      <button
        className="secondary"
        onClick={[Function]}
        type="button"
      >
        View Info
      </button>
    </div>
  </div>
</section>
`;

exports[`render MovieDetailsPage component render <View Info> button clicks correctly 2`] = `
<section
  className="movieDetails"
>
  <figure
    className="inHeroSection"
  >
    <img
      alt="Independance day"
      src={
        Object {
          "canInstrument": true,
          "createTransformer": [Function],
          "getCacheKey": [Function],
          "process": [Function],
        }
      }
    />
  </figure>
  <div
    className="movieInfo"
  >
    <div
      className="movieInfo__left"
    >
      <h1
        className="primary"
      >
        Independance day
      </h1>
      <div
        className="movieSubInfo"
      >
        <ul
          className="primary"
          style={
            Object {
              "order": null,
            }
          }
        >
          <li>
            scifi
          </li>
          <li>
            comedy
          </li>
        </ul>
        <span
          className="runtime"
        >
          4h
        </span>
      </div>
      <div
        className="rating"
      >
        <span
          className="stars"
          style={
            Object {
              "width": "10.4em",
            }
          }
        />
        <span
          className="primary"
        >
          1.9
        </span>
      </div>
    </div>
    <p
      className="primary"
    >
      <span>
        There are growing dangers in
    the wizarding world of 1926 New York.
    Something mysterious is leaving a path of destruction in the streets,
    threatening to expose the wizarding
      </span>
    </p>
    <div
      className="movieInfo__right"
    >
      <button
        className="secondary"
        onClick={[Function]}
        type="button"
      >
        Watch Now
      </button>
      <button
        className="secondary"
        onClick={[Function]}
        type="button"
      >
        View Info
      </button>
    </div>
  </div>
</section>
`;

exports[`render MovieDetailsPage component render component correctly 1`] = `
<section
  className="movieDetails"
>
  <figure
    className="inHeroSection"
  >
    <img
      alt="Independance day"
      src={
        Object {
          "canInstrument": true,
          "createTransformer": [Function],
          "getCacheKey": [Function],
          "process": [Function],
        }
      }
    />
  </figure>
  <div
    className="movieInfo"
  >
    <div
      className="movieInfo__left"
    >
      <h1
        className="primary"
      >
        Independance day
      </h1>
      <div
        className="movieSubInfo"
      >
        <ul
          className="primary"
          style={
            Object {
              "order": null,
            }
          }
        >
          <li>
            scifi
          </li>
          <li>
            comedy
          </li>
        </ul>
        <span
          className="runtime"
        >
          4h
        </span>
      </div>
      <div
        className="rating"
      >
        <span
          className="stars"
          style={
            Object {
              "width": "10.4em",
            }
          }
        />
        <span
          className="primary"
        >
          1.9
        </span>
      </div>
    </div>
    <p
      className="primary"
    >
      <span>
        There are growing dangers in
    the wizarding world of 1926 New York.
    Something mysterious is leaving a path of destruction in the streets,
    threatening to expose the wizarding
      </span>
    </p>
    <div
      className="movieInfo__right"
    >
      <button
        className="secondary"
        onClick={[Function]}
        type="button"
      >
        Watch Now
      </button>
      <button
        className="secondary"
        onClick={[Function]}
        type="button"
      >
        View Info
      </button>
    </div>
  </div>
</section>
`;
